x-pack/test/security_solution_api_integration/config/services/security_solution_serverless_supertest.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// import { FtrProviderContext } from '@kbn/ftrcommonfunctionalservices';
+import supertest from 'supertest';
+import { format as formatUrl } from 'url';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+// It is wrapper around supertest that injects Serverless auth headers
+export async function SecuritySolutionServerlessSuperTest({ getService }: FtrProviderContext) {
+  const config = getService('config');
+
+  const kbnUrl = formatUrl({
+    ...config.get('servers.kibana'),
+    auth: false,
+  });
+
+  const svlUserManager = getService('svlUserManager');
+  const adminRoleAuthc = await svlUserManager.createApiKeyForRole('admin');
+
+  const svlCommonApi = getService('svlCommonApi');
+  const commonRequestHeader = svlCommonApi.getCommonRequestHeader();
+
+  return supertest.agent(kbnUrl).set(commonRequestHeader).set(adminRoleAuthc.apiKeyHeader);
```

Comment:
Why are we always returning the symbol, hardcoded to the admin role?

Reply:
This PR goal is to stop using the operator user. https://github.com/elastic/kibana/issues/183512
We can achieve this goal by using the `admin` role by default without having to manually update every test.

I have updated the code and introduced a service that allows us to create a `supertest` client for a given role.
`const supertest = createSuperTest('admin')`

Due to a bug, it only works for the admin role at the moment. The `svlUserManager.createApiKeyForRole` function fails when the user has no API key creation rights, which fails for most roles. When this bug is solved we will need to manually update all security solution FTR tests to use the appropriate role.

Reply:
May I have a link to the bug?

Reply:
I could only find this slack thread https://elastic.slack.com/archives/C04HT4P1YS3/p1717432409149569

Reply:
I found ithttps://github.com/elastic/kibana/issues/184948